### PR TITLE
fix!: replaces .String() method with internal method .TimeToString().

### PR DESCRIPTION
**Pull request by marcosuma**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): https://jira.mongodb.org/browse/CLOUDP-215154

### Changelog

Fixes date fields that were not compliant to the ISO 8601 timestamp format in UTC, in line with the documentation.

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

**Comment by marcosuma** [comment]
@lantoli @AgustinBettati I am not sure if I missed some specific fields since they weren't documented in the Jira issue.

**Review by lantoli — APPROVED** [review]

**Comment by lantoli** [comment]
> @lantoli @AgustinBettati I am not sure if I missed some specific fields since they weren't documented in the Jira issue.

you're right there is not a list as the task was to look through all the code.   

This only happens in new SDK. But it's not straightforward to know what fields are time.Time as sometimes only the attribute name appears but not the type.

**Review by oarbusi — APPROVED** [review]

File: mongodbatlas/data_source_cloud_shared_tier_restore_job.go [hunk 1]
```diff
@@ -102,23 +103,23 @@ func dataSourceMongoDBAtlasCloudSharedTierRestoreJobsRead(ctx context.Context, d
 		return diag.FromErr(fmt.Errorf("error setting `snapshot_id` for shared tier restore job '%s': %w", jobID, err))
 	}
 
-	if err = d.Set("snapshot_finished_date", job.GetSnapshotFinishedDate().String()); err != nil {
+	if err = d.Set("snapshot_finished_date", conversion.TimeToString(job.GetSnapshotFinishedDate())); err != nil {
```
Comment (andreaangiolillo) [inline mongodbatlas/data_source_cloud_shared_tier_restore_job.go:106]: Why Is this a breaking change 🤔 ? Looking at the old code, it was still using a string. Is the format of the time different?

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Review by AgustinBettati — APPROVED** [review]
LGTM, went over the `.String(` results in the repo and all have been adjusted

File: mongodbatlas/data_source_cloud_shared_tier_restore_job.go (hunk 1 above)
Comment (lantoli) [reply mongodbatlas/data_source_cloud_shared_tier_restore_job.go:106]: +1 , we introduced a bug when migrating to new SDK and here we're fixing it

Comment (AgustinBettati) [reply mongodbatlas/data_source_cloud_shared_tier_restore_job.go:106]: > Is the format of the time different?

yes, it is now aligned with how Atlas API and old SDK format time fields.

I do see the value of marking as a breaking change. If a user has a configuration that parses these attributes, the new version will likely force the user to adapt how the string is being parsed (unlikely given migration to new SDK was also recent, but worth noting in changelog).

Comment (marcosuma) [reply mongodbatlas/data_source_cloud_shared_tier_restore_job.go:106]: > I do see the value of marking as a breaking change. If a user has a configuration that parses these attributes, the new version will likely force the user to adapt how the string is being parsed (unlikely given migration to new SDK was also recent, but worth noting in changelog).

+1 to this, that was my rationale.

**Pull request merged**